stats/opentelemetry/client_metrics.go

```diff
@@ -128,9 +128,13 @@ func (h *clientStatsHandler) streamInterceptor(ctx context.Context, desc *grpc.S
 }
 
 func (h *clientStatsHandler) perCallMetrics(ctx context.Context, err error, startTime time.Time, ci *callInfo) {
-	s := status.Convert(err)
-	callLatency := float64(time.Since(startTime)) / float64(time.Second)
-	h.clientMetrics.callDuration.Record(ctx, callLatency, otelmetric.WithAttributes(otelattribute.String("grpc.method", ci.method), otelattribute.String("grpc.target", ci.target), otelattribute.String("grpc.status", canonicalString(s.Code()))))
+	callLatency := float64(time.Since(startTime)) / float64(time.Second) // calculate ASAP
+	attrs := otelmetric.WithAttributeSet(otelattribute.NewSet(
+		otelattribute.String("grpc.method", ci.method),
+		otelattribute.String("grpc.target", ci.target),
+		otelattribute.String("grpc.status", canonicalString(status.Convert(err).Code())),
```

Comment:
Can we use `status.Code(err)` here?

Reply:
Done. Rebased and addressed this comment. No other changes.

Reply:
For future consideration, please don't force push during github PR reviews -- or if you want to do a rebase, please don't modify the contents of existing commits.  Github doesn't maintain review consistency well in these cases.  It's much easier to see what's changed over time if you only ever add new commits.  Thanks!

Reply:
Ok, got it. Thank you. I mainly do it to have a clean history where each commit makes sense, but if in this repo PRs are squash-merged anyway, then there is no point. I'll keep this in mind. 👍